File: app/src/main/java/com/urt1rs/donkeywiki/entity/employee/Texas.java
package com.urt1rs.donkeywiki.entity.employee;


import com.urt1rs.donkeywiki.constant.employee.Position;
import com.urt1rs.donkeywiki.constant.employee.Tag;

import java.util.ArrayList;

/**
 * 德克萨斯
 */
public class Texas extends Employee {

    private static Texas instance;

    private Texas() {
        position = Position.PIONEER;
        tags = new ArrayList<>(5);
        tags.add(Tag.COST_RESTORE);
    }

    private static Texas getInstance() {
        if (null == instance) {
            synchronized (Texas.class) {
                if (null == instance) {
                    instance = new Texas();
                }
            }
        }
        return instance;
    }



}

File: app/src/main/java/com/urt1rs/donkeywiki/entity/employee/Texas.kt
package com.urt1rs.donkeywiki.entity.employee


import com.urt1rs.donkeywiki.constant.employee.Position
import com.urt1rs.donkeywiki.constant.employee.Tag

import java.util.ArrayList

/**
 * 德克萨斯
 */
class Texas private constructor() : Employee() {

    init {
        position = Position.PIONEER
        tags = ArrayList(5)
        tags!!.add(Tag.COST_RESTORE)
    }

    companion object {

        private var instance: Texas? = null

        private fun getInstance(): Texas {
            if (null == instance) {
                synchronized(Texas::class.java) {
                    if (null == instance) {
                        instance = Texas()
                    }
                }
            }
            return instance!!
        }
    }


}
